File: src/oreo.cpp
#include <string>
#include <iostream>
#include <fstream>
#include <vector>

#include "lexer.hpp"
#include "parser.hpp"

using std::string;

int main(int argc, char **argv) {
    string text;
    string tokenValue = "";
    std::ifstream File(argv[1]);
    std::ofstream NewFile("out/Scribe.txt");

    std::vector<Token> tokens;

    while (!File.eof()) {
        // look ahead until it finds a break character
        text = File.get();

        for (int i = 0; i < text.length(); i++) {
            if (text[i] == ' ' || File.eof()) {
                tokens.push_back(lex(tokenValue));
                tokenValue = "";
                break;
            } else {
                tokenValue.push_back(text[i]);
            }
        }
    }

    // parse happens here
    Parser parser(tokens);
    Node ast = parser.Parse();

    std::cout << "Data: " << ast.right->data << "\n";

    File.close();
    NewFile.close();
}

File: typescript/oreo.ts
import * as fs from 'fs';
import {Token, lex} from './lexer';
import { BinOpNode, NumberNode, Parser} from './parser';

let tokenValue: string = "";
let tokens: Token[] = [];

let file: string = process.argv[2];

const fileContent: string = fs.readFileSync(file, 'utf-8');

for (let i = 0; i <= fileContent.length; i++) {
    if(fileContent.charAt(i) == ' ' || fileContent.charAt(i) == fileContent.charAt(fileContent.length)) {
        tokens.push(lex(tokenValue));
        tokenValue = "";
    } else {
        tokenValue = tokenValue + fileContent.charAt(i);
    }
}

const parser: Parser = new Parser(tokens);
const ast: BinOpNode = parser.Parse();

console.log(ast);